fractal_graphing/img/img.go

```diff
@@ -0,0 +1,114 @@
+// Copyright © 2016 Pawel Rozlach.
+// License: https://creativecommons.org/licenses/by-nc-sa/4.0/
+
+// Based on work by Alan A. A. Donovan & Brian W. Kernighan
+// which can be found at:
+// https://github.com/adonovan/gopl.io.git
+
+// Mandelbrot emits a PNG image of the Mandelbrot fractal.
+
+package img
+
+import (
+	"fmt"
+	"image"
+	"image/color"
+	"os"
+	"runtime"
+	"sync"
+
+	"github.com/vespian/go-excersizes/fractal_graphing/algos"
+	"github.com/vespian/go-excersizes/fractal_graphing/common"
+)
+
+// (sur) One more question about breaking the line (here a func. signature) :)
+func BuildImg(width, height, scaling int,
+	algo func(r, i float64) (uint8, uint8)) *image.RGBA {
+
+	img := image.NewRGBA(image.Rect(0, 0, width, height))
+	wait_group, worker_ch := spawnProcessors(scaling, img, algo)
+
+	for py := 0; py < height; py += common.TILE_SIZE {
+		for px := 0; px < width; px += common.TILE_SIZE {
+			// (sur) should I worry about generating to many objects for GC ?
+			worker_ch <- [2]int{py, px}
+		}
+	}
+
+	close(worker_ch)
+	wait_group.Wait()
+
+	return img
+}
+
+func spawnProcessors(subsample_f int, img *image.RGBA, algo algos.AlgoFunc) (
+	*sync.WaitGroup, chan<- [2]int) {
```

Comment:
1. this newline is strange.
2. returning channels is not nice, pass them from the outside